docs/internals/data_file.md

```diff
@@ -0,0 +1,196 @@
+---
+sidebar_position: 2
+---
+
+# Data File
+
+> “Just show me the tables already!”
+> — probably not Fred Brooks
+
+Each TigerBeetle replica stores all data inside a single file, called the data file (conventional
+extension is `.tigerbeetle`). This document describes the high level layout of the data file. The
+presentation is simplified a bit, to provide intuition without drowning the reader in details.
+Consult the source code for byte-level details!
+
+The data file is divided into several zones, with the main ones being:
+
+- write-ahead log
+- superblock
+- grid
```

Comment:
The current ordering is intentional:

- I want to omit client replies, as I want to focus on essentials, rather than on being the reference
- The order "rhymes" throughout the document, it is wal/superblock/grid grid/superblock/wal wal/superblock/grid grid/superblock/wal, so that it sticks better (and the rhyme is set by the main body of explanation, which is grid->superblock->wal, as that's the most logical order: we start with the grid because that's where we store the data, then we add superblock because we need atomic updates, and then we add wal, because you can't make _every_ update atomic, and need to buffer data somewhere).